Jamulus.pro

```diff
@@ -229,18 +229,42 @@ win32 {
     LIBS += -framework AVFoundation \
         -framework AudioToolbox
 } else:android {
-    ANDROID_ABIS = armeabi-v7a arm64-v8a x86 x86_64
+    # ANDROID_ABIS = armeabi-v7a arm64-v8a x86 x86_64
+    # note: only armeabi-v7a arm64-v8a are targeted, others are dead/deprecated
+
+    # get ANDROID_ABIS from environment - passed directly to qmake
+    ANDROID_ABIS = $$getenv(ANDROID_ABIS)
+
+    # if ANDROID_ABIS is passed as env var to qmake, will override this
+    # !defined(ANDROID_ABIS, var):ANDROID_ABIS = arm64-v8a
+
+    # Experiments show likely better results with Android 10+ devices
+    ANDROID_MIN_SDK_VERSION = 29
+    ANDROID_TARGET_SDK_VERSION = 32
     ANDROID_VERSION_NAME = $$VERSION
-    ANDROID_VERSION_CODE = $$system(git log --oneline | wc -l)
+
+    # date-based unique value - prevent clashes in Play Store build numbering
+    !defined(ANDROID_VERSION_CODE, var):ANDROID_VERSION_CODE = $$system(date +%s | cut -c 2-)
```

Comment:
This will not work on windows I suppose? Wouldn't the git commit hash or something else be better?

Reply:
The CI is obviously targeted purely for Linux.
For local building on Windows we can just hardcode any int value.
`ANDROID_VERSION_CODE = 123456`

Android Version Code requires an integer <=2100000000.
So you'd need some custom logic to convert the shorthash to int first.

I don't like relying on git log because you need a separate version code for armeabi7 and armv8 builds.
The above `date` output guarantees a unique, valid value for each Play Store submission.

Reply:
> The CI is obviously targeted purely for Linux.

But as it's in Jamulus.pro we should ensure cross compatibility. Even if it's a noop.